### PR TITLE
[v2] ASoC: rt700/rt711: init jack detection after codec is attached

### PR DESCRIPTION
**Pull request by shumingfan**
[v2] ASoC: rt700: init jack detection after rt700 is attached

[v2] ASoC: rt711: init jack detection after rt711 is attached

### CODE REVIEW

File: sound/soc/codecs/rt700.c [hunk 1]
```diff
@@ -330,40 +330,61 @@ static void rt700_btn_check_handler(struct work_struct *work)
...
+	}
 
 	/* power off */
 	if (dapm->bias_level <= SND_SOC_BIAS_STANDBY)
 		regmap_write(rt700->regmap,
 			RT700_SET_AUDIO_POWER_STATE, AC_PWRST_D3);
 
+	dev_dbg(&rt700->slave->dev, "in %s done\n", __func__);
+	mod_delayed_work(system_power_efficient_wq,
+			&rt700->jack_detect_work, msecs_to_jiffies(250));
```
Comment (plbossart) [inline sound/soc/codecs/rt700.c]: do you still want to execute this if the rt700->hs_jack is NULL?

File: sound/soc/codecs/rt700.c [hunk 2]
```diff
@@ -330,40 +330,61 @@ static void rt700_btn_check_handler(struct work_struct *work)
...
+		regmap_write(rt700->regmap,
+			RT700_SET_INLINE_UNSOLICITED_ENABLE, 0x00);
+	}
 
 	/* power off */
 	if (dapm->bias_level <= SND_SOC_BIAS_STANDBY)
 		regmap_write(rt700->regmap,
 			RT700_SET_AUDIO_POWER_STATE, AC_PWRST_D3);
 
+	dev_dbg(&rt700->slave->dev, "in %s done\n", __func__);
```
Comment (plbossart) [inline sound/soc/codecs/rt700.c]: traces should be removed or moved somewhere useful?

File: sound/soc/codecs/rt711.c [hunk 3]
```diff
@@ -399,45 +399,65 @@ static void rt711_btn_check_handler(struct work_struct *work)
...
+			RT711_SET_INLINE_UNSOLICITED_ENABLE, 0x00);
+	}
 
 	/* power off */
 	if (dapm->bias_level <= SND_SOC_BIAS_STANDBY)
 		regmap_write(rt711->regmap,
 			RT711_SET_AUDIO_POWER_STATE, AC_PWRST_D3);
 	mutex_unlock(&rt711->calibrate_mutex);
 
+	dev_dbg(&rt711->slave->dev, "in %s done\n", __func__);
```
Comment (plbossart) [inline sound/soc/codecs/rt711.c]: trace is not very useful, should either remove or move it up.

File: sound/soc/codecs/rt711.c [hunk 4]
```diff
@@ -399,45 +399,65 @@ static void rt711_btn_check_handler(struct work_struct *work)
...
+	mod_delayed_work(system_power_efficient_wq,
+			&rt711->jack_detect_work, msecs_to_jiffies(250));
```
Comment (plbossart) [inline sound/soc/codecs/rt711.c]: do we need to do this when rt711->hs_jack is NULL?

File: sound/soc/codecs/rt711.c [hunk 5]
```diff
@@ -399,45 +399,65 @@ static void rt711_btn_check_handler(struct work_struct *work)
...
-	dev_dbg(&rt711->slave->dev, "in %s...\n", __func__);
+	if (!rt711->hw_init) {
+		dev_dbg(&rt711->slave->dev,
+			"%s hw_init not ready yet\n", __func__);
+		return -EAGAIN;
+	}
 
-	mod_delayed_work(system_power_efficient_wq,
-			   &rt711->jack_detect_work, msecs_to_jiffies(250));
+	dev_dbg(&rt711->slave->dev, "in %s...\n", __func__);
```
Comment (plbossart) [inline sound/soc/codecs/rt711.c]: trace is not very useful

File: sound/soc/codecs/rt700.c [hunk 6]
```diff
@@ -330,40 +330,61 @@ static void rt700_btn_check_handler(struct work_struct *work)
...
-	dev_dbg(&rt700->slave->dev, "in %s...\n", __func__);
+	if (!rt700->hw_init) {
+		dev_dbg(&rt700->slave->dev,
+			"%s hw_init not ready yet\n", __func__);
+		return 0;
+	}
 
-	mod_delayed_work(system_power_efficient_wq,
-			   &rt700->jack_detect_work, msecs_to_jiffies(250));
+	dev_dbg(&rt700->slave->dev, "in %s...\n", __func__);
```
Comment (plbossart) [inline sound/soc/codecs/rt700.c]: trace isn't very useful

**Comment by shumingfan** [comment]
> @shumingfan looks good, couple of questions below.
> Also you'd need to rebase, I changed the code and this creates conflicts I think.

Thanks for review. I will resend the patches.

**Pull request closed without merging**